alibi_detect/cd/metrics.py

```diff
@@ -0,0 +1,333 @@
+import numpy as np
+from typing import Union, List, Optional, Tuple, Callable
+import matplotlib.pyplot as plt
+from scipy.stats import uniform
+import statsmodels.api as sm
+import seaborn as sns
+#  import warnings
+from tqdm import tqdm
+
+
+def eval_calibration(
+        detector: type,  # TODO: type w/ BaseDriftDetector if introduced.
+        X: Union[np.ndarray, list],
+        C: Optional[np.ndarray] = None,
+        model: Optional[Callable] = None,
+        detector_kwargs: Optional[dict] = None,
+        n_runs: int = 100,
+        n_samples: int = 500,
+        qq_plot: bool = False,
+        hist_plot: bool = False,
+        random_seed: Optional[Union[int, np.random.Generator]] = None
+) -> np.ndarray:
+    """
+    TODO
+
+    Parameters
+    ----------
+    detector
+        The detector class to evaluate calibration for.
+    X
+        A reservoir of data. On each run instances are randomly subsampled to provide reference and test sets for
+        the detector.
+    C
+        A reservoir of context data. On each run instances are randomly subsampled to provide reference and test sets
+        for the detector. Should only be given for the :class:`~alibi_detect.cd.ContextMMDDrift` detector.
+    model
+        A PyTorch, TensorFlow or Sklearn model. Should only be given for detectors accepting a `model` arg, such
+        as :class:`~alibi_detect.cd.ClassifierDrift`.
+    detector_kwargs
+        Keyword arguments to pass to detector.
+    n_runs
+        The number of experiment runs. A larger number will give more stable p-value distributions, at the expense of
+        a longer runtime.
+    n_samples
```

Comment:
Could we perhaps have separate args for ref and test here? Personally if I was using it I'd want n_test_samples = W and n_ref_samples = N-W

Reply:
`n_samples` is now a tuple `(n_ref_samples, n_test_samples)` like in `eval_test_power`. Does that work?

Reply:
Sounds good